<launch>
  <!-- Allow overriding the world from the command line -->
  <arg name="world_name" default="iaslab_assignment1"/>

  <!-- 1) Start the simulation with the requested world -->
  <include file="$(find tiago_iaslab_simulation)/launch/start_simulation.launch">
    <arg name="world_name" value="$(arg world_name)"/>
  </include>

  <!-- 2) Navigation stack -->
  <include file="$(find tiago_iaslab_simulation)/launch/navigation.launch"/>

  <!-- 3) AprilTag pipeline (use apriltag.launch as you listed) -->
  <include file="$(find tiago_iaslab_simulation)/launch/apriltag.launch"/>

  <!-- 4) AprilTags IDs server -->
  <node pkg="tiago_iaslab_simulation"
        type="apriltag_ids_generator_node"
        name="apriltag_ids_generator_node"
        output="screen"/>

</launch>
